AWS_LAMBDA/remove_predictions_from_exited_guests/lambda_function.py

```diff
@@ -0,0 +1,69 @@
+# Each day, guests are exiting. Once a guest has exited, the predicted_exit_destination
+# column should be "exited" rather than a prediction.
+
+import os
+import sys
+import logging
+import psycopg2
+
+# RDS settings
+rds_host = os.environ.get('RDS_HOST')
+rds_username = os.environ.get('RDS_USERNAME')
+rds_user_pwd = os.environ.get('RDS_USER_PWD')
+
+# Logging
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+# Connect to database
+try:
+    # If able to connect
+    conn = psycopg2.connect(
+        host=rds_host,
+        user=rds_username,
+        password=rds_user_pwd)
+except:
+    # If unable to connect
+    logger.error("ERROR: Could not connect to Postgres instance.")
+    sys.exit()
+
+# Log connection success
+logger.info("SUCCESS: Connection to RDS Postgres instance succeeded")
+
+
+def lambda_handler(event, context):
+    '''Add predicted_exit_destination column to guests_temp if not exists, 
+    then updates guests_temp to account for newly exited guests (if exited, no more need for prediction on row).'''
+
+    # Check if column exists
+    if conn != None: 
+        cur = conn.cursor()
+
+        try:
+            # If column exists
+            query = "SELECT predicted_exit_destination FROM guests_temp;"
+            cur.execute(query)
+            
+        except:
+            conn.rollback()
+            # If column doesn't exist
+            query = "ALTER TABLE guests_temp ADD predicted_exit_destination VARCHAR;"
```

Comment:
Better yet, we could remove this try/except block and instead use the IF EXISTS.

See: https://stackoverflow.com/questions/12597465/how-to-add-column-if-not-exists-on-postgresql

`ALTER TABLE table_name ADD COLUMN IF NOT EXISTS column_name INTEGER;`

Ideally, we should be doing all the logic of this entire script inside on a single query. Something similar to the first answer in that stackoverflow thread.

Reply:
I had tried to find the syntax for "adding if not exists" like I suggested last time we met. However, I wasn't able to find the answer at the time and this is what I came up with and sent to you Thursday as far as I know.

Reply:
Added IF NOT EXISTS to first query, and tested it with database. Take a look and let me know what you think!

Reply:
https://github.com/nonprofit-intake/data_engineering/pull/3/files#diff-4c41ec8ab5636da80c743a72884ea467750616b3aa6177ceafde7319961369ec

Reply:
https://github.com/nonprofit-intake/data_engineering/blob/b3a1f161051d3d753c0f5cae6aa4c615d6f3ffc4/AWS_LAMBDA/remove_predictions_from_exited_guests/lambda_function.py

Reply:
Beautiful! Looks so much cleaner. I tested both queries with all the cases I could think of and they're working perfectly. :100: